### PR TITLE
[CSPM] Handle system characteristics in XCCDF evaluator

### PR DESCRIPTION
**Pull request by 0intro**
### What does this PR do?

This change adds a function to OpenSCAP to print
system characteristics. This function is called
from the oscap-io tool when passing the "-syschar" parameter.

The XCCDF evaluator has been modified to pass the
"-syschar" parameter when calling the oscap-io tool and parse the system characteristics output.

The system characteristics are sent to the backend as part of the data array.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/compliance/scap/syschar.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+			if object.Flag == "complete" {
+				if doc.OvalSystemCharacteristics.SystemData != nil {
+					for _, reference := range object.Reference {
+						for _, item := range doc.OvalSystemCharacteristics.SystemData.Item {
+							if reference.ItemRef != item.Id {
+								continue
+							}
+							i := Item{
+								ID:       item.XMLName.Local,
+								Messages: make(map[string]string, 0),
```
Comment (jinroh) [inline pkg/compliance/scap/syschar.go:56]: you can pre-allocate the map:

```go
make(map[string]string, len(tem.Message))
```

File: pkg/compliance/evaluator_xccdf.go [hunk 2]
```diff
@@ -21,13 +21,18 @@ import (
...
+	"github.com/DataDog/datadog-agent/pkg/compliance/scap"
 	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/util/executable"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 	"github.com/DataDog/datadog-agent/pkg/version"
 	"github.com/DataDog/datadog-go/v5/statsd"
 )
 
+const (
+	enableSysChar = true
```
Comment (jinroh) [inline pkg/compliance/evaluator_xccdf.go:33]: let's disable for now.

**Review by jinroh — APPROVED** [review]

File: pkg/compliance/scap/syschar.go (hunk 1 above)
Comment (0intro) [reply pkg/compliance/scap/syschar.go]: Done.

File: pkg/compliance/evaluator_xccdf.go (hunk 2 above)
Comment (0intro) [reply pkg/compliance/evaluator_xccdf.go]: Done.

**Pull request merged**